lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -2,6 +2,8 @@
 
 class SignupsController extends EntityAPIController {
 
+  protected $ignore_global_user = FALSE;
```

Comment:
If you look at the overridden `save()` method in the controller, you'll see that it compares passed in [user with global user](https://github.com/DoSomething/phoenix/pull/7018/files#diff-d06deaedbb4cf86687781d67ba7d5eddR120) regardless and will always if it's coming from the magic link and global user thus anon.